### DISCUSSION TITLE
How to populate error prop in functions.invoke response?

### DISCUSSION THREAD

**Discussion by timdiacon** [supabase/supabase#9126] [Questions]
When calling an edge function via `supabaseClient.functions.invoke` the response can be destructured to `{ data, error }` but I can't find any documentation on how to populate the error prop from the edge function? I'm sending back a Response object as per the documentation, whatever is placed in the body of the response populates the `data` prop on the front end. Is there a way of populating the error prop instead somehow?

`return new Response(JSON.stringify(error), {
   status: 500
})`

**Comment by sdaoud** [comment]
Supabase functions use Deno's [Response](https://deno.com/deploy/docs/runtime-response#properties) interface. We can send back data to match `{ data, error }` by doing something like this:

```js
const { data, error } = await something()

if (error) {
  return new Response(JSON.stringify({ error: error }), { status: 500 })
} else {
  return new Response(JSON.stringify({ data: data }), { status: 200 })
}
```

Or even better, if we know `something()` always returns `{ data, error }` we can do:

```js
const response = await something()
return new Response(JSON.stringify(response), { status: response.error ? 500 : 200 })
```

**Reply by sdaoud** [reply]
When you're calling `return new Response(JSON.stringify(error), { status: 500 })` it must be populating the `data` prop because of the contents of `error`. If you do `console.log(error)` before that, I imagine you'll see something like this:

```js
{
  data: {
    message: "something went wrong",
    status: 422
  }
}
```

So you'd need to do this:

```js
if (error) {
  return new Response(JSON.stringify({ error: error.data.message }), { status: error.data.status })
}
```

**Reply by laktek** [reply]
Yes, that does the trick. Check [Supabase's Edge Functions examples](https://github.com/supabase/supabase/tree/master/examples/edge-functions/supabase/functions) for similar implementations:

**Reply by t-lock** [reply]
@laktek Apologies for tagging you on an old thread, but that really doesn't do the trick. When using the example code from the supabase edge functions example, `supabaseClient.functions.invoke()` will always use its own internal error messaging "Edge Function returned a non-2xx status code"...

If you want to pass your own message, it does not seem possible to do so on a Deno response of status of anything other than 200. We have to just give a successful response and attach the error data there.

In Edge function:
```
  try {
    throw new Error("Custom Error Message");
  } catch (error) {
    return new Response(
      JSON.stringify({ error: error.message }),
      {
        headers: { ...corsHeaders, "Content-Type": "application/json" },
        status: 403,
      },
    );
  }
```

In browser client:
```
      const { data, error} = await supabase.functions.invoke("some-function");
      console.log(data, error);
```

The `error` variable logged to console will be "Function returned a non-2xx status code", even though our network tab will show the 403 response with our custom error message, and `data` will be `null`.

However, if we change our status to 200, we will be able to find our error message in `data`: `{error: "Custom Error Message"}`, while `error` will be null.


So the question remains: How can we surface the error message when using a Deno response with a proper error status (non 2xx)?

**Reply by Mohammed-Yasin-Mulla** [reply]
I do have the same question @t-lock .
